dev/sbt-plugin/src/sbt-test/sbt-plugin/projections-happy-path/test

```diff
@@ -0,0 +1,33 @@
+# Structure of this test:
+# =======================
+
+# This tests ensures some java code can manually start a single projection
+# programmatically overriding the `auto-start` flag (see application.conf).
+
+# There are two projections (read-side processor) over a single Persistent
+# entity. A single endpoint returns data produced by both projections. Originally
+# the value projected on `StartedProcessor` is "default-projected-message" but
+# eventually, the status is propagated, the worker is started and the event is
+# processed causing the value to become "Hi".
+
+## First, let's assert everything works as expected in tests
+
+> test
```

Comment:
I don't see the added value of having duplicating the tests. We don't need to assert it twice.
I didn't check all the existing scripted tests, but I don't recall to have seen this mix.

Reply:
I want to see, from a user perspective, that projections work as expected. Not having such tests lead to https://github.com/lagom/lagom/issues/2192 .

Reply:
But the scripted tests are covering that as well, no?

Reply:
@ignasi35, you marked it as a resolved conversation, but maybe you didn't see my last comment.

The scripted tests are already covering issue #2192, so why the duplication?

Reply:
btw, if we want to cover issue #2192 with such a test, we should have it on the project itself. Much easier to debug and faster to run then on a scripted test.

Reply:
`scripted` tests provide what's closest to end-user feel. There's no shortcuts or package-visibility holes: you create end-user code and it works.

My idea with these scripted tests is not so much to test the sbt plugin as it is to test that from an end-user codebase everything works as expected. On a unit-test, users create a single-node server with multiple hacks (e.g. no service locator). We don't have any place to write integration tests that mix `MyLoader` (where a `Components` cake is composed) with fully functional persistence and cluster distribution other than scripted tests.

`scripted` tests are a perfect environment to simulate dev mode and service-level unit tests (aka, integration tests) ensuring the `Components` and `TestService` we provide are not getting on the way.

Reply:
I think the `multi-jvm` tests for `ProjectionRegistry` are really good until we can decouple `ProjectionRegistryActor` from `Replicator`. When we achieve that decoupling we'll even be able to unit test `ProjectionRegistryActor`.

Reply:
> you marked it as a resolved conversation

Apologies, I was trying to mark the other one and probably miss-clicked (?). :-(

Reply:
I still don't see why this justify the duplication. What can not be tested in the scripted test that is being tested on `HelloServiceTest`?

Reply:
We have some tests like that in Play at least:

- https://github.com/playframework/playframework/blob/8190355ba29be250847755bdc09ca63752cf8eb3/dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/http-backend-netty/test#L4
- https://github.com/playframework/playframework/blob/8190355ba29be250847755bdc09ca63752cf8eb3/dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/http-backend-akka-http-http2/test#L4
- https://github.com/playframework/playframework/blob/8190355ba29be250847755bdc09ca63752cf8eb3/dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/shutdown-happy-path/test#L58-L59

And the idea there is what @ignasi35 explained: test more closely from what a user will see in a way we cannot when using unit/integration/multi-jvm tests. But I see your point, @renatocaval. In general I would prefer to avoid scripted tests as much as possible (for multiple reasons), but I don't see how here. Do you think it is possible to have the same test confidence with an integration test?

Reply:
> What can not be tested in the scripted test that is being tested on HelloServiceTest?

The was a cluster is built, or how a topic producer is started (ok, topic producers are not exercised in these particular tests) differs. Just so slightly and yet enough that's worth it. IMHO

Reply:
@marcospereira, the examples you posted are indeed running `test`, but that that's all. 
Here we are running `test` and then `runAll` and executing exactly the same test, but then using the external API. We are exercising it twice.

* here we run `test` -> https://github.com/lagom/lagom/pull/2231/files#diff-022af7f20da34e85d96cd06d66fb33ffR15
* here we start the all -> https://github.com/lagom/lagom/pull/2231/files#diff-022af7f20da34e85d96cd06d66fb33ffR19
* and here we run exactly the same assertions ->  https://github.com/lagom/lagom/pull/2231/files#diff-022af7f20da34e85d96cd06d66fb33ffR22-R31

I was not saying to avoid the scrpited test. My point is that we have twice the same test, one using the testkit and one using script. 

Both tests are exercising exactly the same. Both tests are running in an artificial setup (dev-mode and testkit). And with respect to issue #2192, both are covering it because in both cases we need to build the cake. 

I'm ok to have some duplication if there is a clear reason for it, but I still don't see the reason here. Sorry @ignasi35, I don't think your last response answered it. I don't get the point about the cluster and the topic producer argument you debunked yourself. 

dev-mode is closer to prod, closer than testkit, so I'm in favour of removing the duplication and keep only the scripted (runAll + assertions).

Unless there is a case that we are not covering with scripted test. In that case, please explicitly indicate what it is.

Reply:
@renatocaval now I see. I cannot understand the duplication either.

@ignasi35, do you think it makes sense to have just the clustered tests exercised by `test` and have the other things tested by `runAll` and `assertRequest` commands?

Reply:
@marcospereira, there is no difference with respect to clustering. Both cases are forming a single node cluster.

Reply:
I had a chat with @ignasi35 on this. We are going to keep as-is for now, but we plan to have this kind of integration test moved out of scripted test. Meaning that `HelloServiceTest` will move out. The real scripted test can be then refactored to a much smaller 'happy path' test.